commcare_connect/opportunity/models.py

```diff
@@ -408,9 +408,16 @@ class CompletedWork(models.Model):
     reason = models.CharField(max_length=300, null=True, blank=True)
     status_modified_date = models.DateTimeField(null=True)
 
-    def update_status(self, status):
-        self.status = status
+    def __init__(self, *args, **kwargs):
+        self.status = CompletedWorkStatus.incomplete
```

Comment:
Doesn't the `default=CompletedWorkStatus.incomplete` on [model def](https://github.com/dimagi/commcare-connect/pull/359/files#diff-df90e317de0ae052504b3e88cc61aa8cf379effcb469ea2f74ac7357ac42832cR402) ensure this is set by default?

Reply:
@sravfeyn When the values are initialised, the setattr method is used to set those values. During this process, the setattr method attempts to access the property (as implemented). However, since the property (like status) is not present during initialisation, it will throw an error. This behaviour is caused by the Django model.